ARTstor.js

```diff
@@ -34,98 +37,1043 @@
+                            url: urlPDF,
+                            mimeType: "application/pdf"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+                case "21":
+                    var serviceURL4PPT = serverURL + "secure/imagefpx/" + item + "/21"; //PPT
+                    Zotero.Utilities.HTTP.doGet(serviceURL4PPT, function(text) {
+                        var jsonPPT = eval('(' + text + ')');
+                        var urlPPT = server + "thumb" + jsonPPT.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor PowerPoint Attachment",
+                            url: urlPPT,
+                            mimeType: "application/powerpoint"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+                case "22":
+                    var serviceURL4DOC = serverURL + "secure/imagefpx/" + item + "/22"; //DOC
+                    Zotero.Utilities.HTTP.doGet(serviceURL4DOC, function(text) {
+                        var jsonDOC = eval('(' + text + ')');
+                        var urlDOC = server + "thumb" + jsonDOC.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor Word Attachment",
+                            url: urlDOC,
+                            mimeType: "application/msword"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+                case "23":
+                    var serviceURL4XXLS = serverURL + "secure/imagefpx/" + item + "/23"; //XLS
+                    Zotero.Utilities.HTTP.doGet(serviceURL4XXLS, function(text) {
+                        var jsonXLS = eval('(' + text + ')');
+                        var urlXLS = server + "thumb" + jsonXLS.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor Excel Attachment",
+                            url: urlXLS,
+                            mimeType: "application/excel"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+                case "24":
+                    var serviceURL4VID = serverURL + "secure/imagefpx/" + item + "/24"; //Video
+                    Zotero.Utilities.HTTP.doGet(serviceURL4VID, function(text) {
+                        var jsonVID = eval('(' + text + ')');
+                        var urlVID = jsonVID.imageUrl;
+                        artItem.attachments.push({
+                            title: "Artstor Video File Attachment",
+                            url: urlVID,
+                            mimeType: "text/html"
+
+                        });
+                    });
+                    artItem.url = url;
+                    break;
+            }
```

Comment:
most of these are almost identical. Couldn't you create a map between `objectTypeId` and the variable parts? something like
```
var typeMap {
  '12': {
    id: 10,
    mimeType: 'audio/mpeg3',
    attachmentTitle: '....'
  }
}
```
then use that with a single piece of code